src/vs/workbench/contrib/tasks/browser/runAutomaticTasks.ts

```diff
@@ -129,30 +130,29 @@ export class RunAutomaticTasks extends Disposable implements IWorkbenchContribut
 	}
 
 	public static async promptForPermission(taskService: ITaskService, storageService: IStorageService, notificationService: INotificationService, workspaceTrustManagementService: IWorkspaceTrustManagementService,
-		openerService: IOpenerService, workspaceTaskResult: Map<string, IWorkspaceFolderTaskResult>) {
+		openerService: IOpenerService, configurationService: IConfigurationService, workspaceTaskResult: Map<string, IWorkspaceFolderTaskResult>) {
 		const isWorkspaceTrusted = workspaceTrustManagementService.isWorkspaceTrusted;
 		if (!isWorkspaceTrusted) {
 			return;
 		}
-
-		const isFolderAutomaticAllowed = storageService.getBoolean(ARE_AUTOMATIC_TASKS_ALLOWED_IN_WORKSPACE, StorageScope.WORKSPACE, undefined);
-		if (isFolderAutomaticAllowed !== undefined) {
+		const hasShownPromptForAutomaticTasks = storageService.getBoolean(HAS_PROMPTED_FOR_AUTOMATIC_TASKS, StorageScope.WORKSPACE, undefined);
+		if (configurationService.getValue('task.allowAutomaticTasks') !== 'auto' || hasShownPromptForAutomaticTasks !== undefined) {
```

Comment:
I don't think this logic is correct? Don't we need to consider `ALLOW_AUTOMATIC_TASKS` here? Also I don't think `task.allowAutomaticTasks: on` is being handled right?

Reply:
We only want to prompt if the setting value is auto, correct?

Reply:
So show the prompt if it hasn't been shown for this workspace and the setting is auto

Reply:
I haven't checked where this is called from but lower in the function the task is run, so it looks like the task wouldn't be run if it's `on`. This part:

```
if (allow) {
	RunAutomaticTasks._runTasks(taskService, tasks);
}
```

Reply:
You might be right. Too hard to look on a phone. I'll look tomorrow 👍🏻